src/java.base/share/classes/sun/nio/ch/SocketChannelImpl.java

```diff
@@ -403,6 +405,20 @@ private void throwConnectionReset() throws SocketException {
 
     @Override
     public int read(ByteBuffer buf) throws IOException {
+        if (!SocketReadEvent.enabled()) {
```

Comment:
The read/write with sun.nio.ch.SocketInputStream and SocketOutputStream does not go through SC.read/write so I think SocketAdaptor read/write will need attention, maybe a future PR as there are other code paths that aren't covered in this PR.

Reply:
I've created https://bugs.openjdk.org/browse/JDK-8310978 to drive the future PR to support the missing code paths